rest_framework/serializers.py

```diff
@@ -1430,15 +1430,18 @@ def get_unique_together_constraints(self, model):
         """
         for parent_class in [model] + list(model._meta.parents):
             for unique_together in parent_class._meta.unique_together:
-                yield unique_together, model._default_manager
+                yield unique_together, model._default_manager, []
             for constraint in parent_class._meta.constraints:
                 if isinstance(constraint, models.UniqueConstraint) and len(constraint.fields) > 1:
-                    yield (
-                        constraint.fields,
-                        model._default_manager
-                        if constraint.condition is None
-                        else model._default_manager.filter(constraint.condition)
-                    )
+                    if constraint.condition is None:
+                        queryset = model._default_manager
+                        condition_fields = []
+                    else:
+                        queryset = model._default_manager.filter(constraint.condition)
+                        condition_fields = [
+                            f[0].split("__")[0] for f in constraint.condition.children
```

Comment:
not sure how reliable this method of fields extraction

Reply:
I duobt that this will work when the condition uses a cross-reference between two fields like: `Q(first_field__gte=F('second_field'))`.

Reply:
These case should be handled now